src/Task/SymlinkVipGoDir.php

```diff
@@ -86,25 +87,23 @@ public function run(Io $io, TaskConfig $taskConfig): void
             $this->directories->vipMuPluginsDir() => "{$contentDirPath}/mu-plugins",
         ];
         foreach ($this->directories->toArray() as $dirPath) {
-            $map[$dirPath] =  "{$contentDirPath}/" . basename($dirPath);
+            $map[$dirPath] = "{$contentDirPath}/" . basename($dirPath);
         }
 
         $isWindows = Platform::isWindows();
 
         foreach ($map as $target => $link) {
-            if (is_dir($link)) {
-                $this->filesystem->removeDirectory($link);
-            }
```

Comment:
The reason for this was to make sure `$link` was not pointing a _file_, which would cause `$this->filesystem->removeDirectory` to fail.

Considering that `Filesystem::removeDirectory()` would throw an exception in case of any failure, maybe we could always call it (without `if (is_dir($link))`), but wrapping it in `try`/`catch`, something like this:

```php
            try {
                $this->filesystem->removeDirectory($link);
            } catch (\Throwable $throwable) {
                $io->errorLine("Failed creating relative symlink for {$link} to {$target}.");
                $io->errorLine($throwable->getMessage());
                continue;
            }
```

Maybe the `try`/`catch` block could be used to wrap all other operations, so the entire `foreach` could look like this (including the other comment):

```php
        foreach ($map as $target => $link) {
            $errorMsg = "Failed creating relative symlink for {$link} to {$target}.";
            try {
                $this->filesystem->removeDirectory($link);
                $this->filesystem->ensureDirectoryExists($target);
                $link = $this->filesystem->normalizePath($link);

                if ($isWindows) {
                    $this->filesystem->junction($target, $link);
                    continue;
                }

                if (!$this->filesystem->relativeSymlink($target, $link)) {
                    $io->errorLine($errorMsg);
                }
            } catch (\Throwable $throwable) {
                $io->errorLine($errorMsg);
                $io->errorLine($throwable->getMessage());
                continue;
            }
        }
```

Reply:
Hm...i agree, that this shouldn't be just a comment and an error if the creating of a symlink does not succeed. 
My first intention with that Pull Request was to get  more feedback from the script when something does not work as expected. I thought, that this is not directly an error, since we're currently just executing everything without stopping the script. 

The symlink-task is important to succeed. We might should consider not catching Exceptions inside every Task and let them bubble up to the Runner, so that he can take care of it. Actually every Task is required to succeed, to have a working system. So on an Exception or an error (which _could_ throw a custom Exception) we should stop the program with a meaningful error message. :)